tests/extensions/tabular/test_xlsx_tools.py

```diff
@@ -17,4 +19,34 @@ def test_xlsx_xlrd(self):
         assert sheet[0][2] == {'field': 'three', 'id': 'three', 'name': 'three', 'sortable': True}
         assert sheet[1][0] == {'one': 'a', 'two': 'b', 'three': 'c'}
         assert sheet[1][1] == {'one': 1.0, 'two': 2.0, 'three': 3.0}
-        assert sheet[1][2] == {'one': u'wierd\\x97', 'two': u'char\\x98','three': u'set\\x99'}
+        assert sheet[1][2] == {'one': u'wierd\\x97', 'two': u'char\\x98', 'three': u'set\\x99'}
+
+    def test_xlsx_xlrd_duplicate_fields(self):
+        with open(os.path.join(BASE, 'files', 'test_duplicate.xlsx')) as fp:
+            sheets = xlrd_tools.xlsx_xlrd(fp)
+
+        sheet = sheets.popitem()[1]
+        assert sheet[0][0] == {'id': 'Name', 'name': 'Name', 'field': 'Name', 'sortable': True}
+        assert sheet[0][1] == {'id': 'Dup (1)', 'name': 'Dup (1)',
+                                'field': 'Dup (1)', 'sortable': True}
+        assert sheet[0][2] == {'id': 'Dup (2)', 'name': 'Dup (2)',
+                                'field': 'Dup (2)', 'sortable': True}
+        assert sheet[0][3] == {'id': 'Dup (3)', 'name': 'Dup (3)',
+                                'field': 'Dup (3)', 'sortable': True}
+        assert sheet[0][4] == {'id': 'Dup (4)', 'name': 'Dup (4)',
+                                'field': 'Dup (4)', 'sortable': True}
+        assert sheet[0][5] == {'id': 'Not Dup', 'name': 'Not Dup',
+                                'field': 'Not Dup', 'sortable': True}
+        assert sheet[1][0] == {'Name': 1.0, 'Dup (1)': 2.0, 'Dup (2)': 3.0,
+                            'Dup (3)': 4.0, 'Dup (4)': 5.0, 'Not Dup': 6.0}
+
+    # After demo it was suggested the iteration cap be raised. The value ended up to be about 5,000
```

Comment:
As suggested above, use a default arg for `iterations` and set it lower. You can then use this for this test instead of having to make a file to iterate 5000 times.